src/automaton.jl

```diff
@@ -0,0 +1,78 @@
+abstract type Automaton end
+
+mutable struct AutomatonList <: Automaton
+    nstates::Int
+    nsymbols::Int
+    transitions::Vector{Tuple{Int, Int, Int}}
+    issorted::Bool
+end
+
+function NewAutomatonList(nstates, nsymbols)
+    transitions = Tuple{Int, Int, Int}[]
+    return AutomatonList(nstates, nsymbols, transitions, true)
+end
+
+function ensure_sorted!(autom::AutomatonList)
+    if !autom.issorted
+        # display("autom not sorted")
+        sort!(autom.transitions)
+        autom.issorted = true
+    end
+end
+
+function get_ntrans(autom::AutomatonList)
+    return length(autom.transitions)
+end
+
+# Do not check that source, symbol, target are "inbounds"
+# Assumes not add twice same transition...
+function add_transition!(autom::AutomatonList, source, symbol, target)
+    push!(autom.transitions, (target, source, symbol))
+    autom.issorted = false
+end
+
+function Base.empty!(autom::AutomatonList)
+    empty!(autom.transitions)
+    autom.issorted = true
+end
+
+function compute_post!(targetlist, autom::AutomatonList, source, symbol)
+    ensure_sorted!(autom)
+    check = x -> x[2] == source && x[3] == symbol
+    for trans in Iterators.filter(check, autom.transitions)
```

Comment:
Why not simply
```julia
for trans in autom.transitions
    if trans[2] == source && trans[3] == symbol
        ...
```
It would allocate less than `filter`